pkg/ipamd/ipamd.go

```diff
@@ -1332,15 +1342,27 @@ func (c *IPAMContext) SetNodeLabel(key, value string) error {
 	}
 
 	// Update node status to advertise the resource.
-	_, err = c.k8sClient.CoreV1().Nodes().Update(updateNode)
+	err = c.k8sClient.Update(ctx, updateNode)
 	if err != nil {
 		log.Errorf("Failed to update node %s with label %q: %q, error: %v", c.myNodeName, key, value, err)
 	}
-	log.Infof("Updated node %s with label %q: %q", c.myNodeName, key, value)
+	log.Debugf("Updated node %s with label %q: %q", c.myNodeName, key, value)
+
 	return nil
 }
 
 // GetPod returns the pod matching the name and namespace
-func (c *IPAMContext) GetPod(podName, namespace string) (*v1.Pod, error) {
-	return c.k8sClient.CoreV1().Pods(namespace).Get(podName, metav1.GetOptions{})
+func (c *IPAMContext) GetPod(podName, namespace string) (*corev1.Pod, error) {
+	ctx := context.Background()
+	var pod corev1.Pod
+
+	podKey := types.NamespacedName{
+		Namespace: namespace,
+		Name:      podName,
+	}
+	err := c.standalonek8sClient.Get(ctx, podKey ,&pod)
```

Comment:
Why doesn't this use the cached client?

Reply:
We wanted to avoid a cached client for Pods(essentially avoid a watch on pod resource from every node) and so we're using a Cached client exclusively for `ENIConfigs` and `Nodes` and depend on raw client to get `Pod` objects.